public_html/wp-content/plugins/wordcamp-organizer-survey/includes/debrief-survey/cron.php

```diff
@@ -0,0 +1,183 @@
+
+			update_option( TEMP_ATTENDEE_ID, $attendees_id );
+			log( 'Successfully added attendee:', get_email_id(), $attendees_id );
+		} else {
+			log( 'Failed to add attendee:', get_email_id(), $attendees_id );
+		}
+	}
+}
+
+/**
+ * Associates attendee to emails.
+ */
+function associate_attendee_to_email( $email_id ) {
+	$recipient_id = get_wordcamp_attendees_id();
+
+	if ( empty( $recipient_id ) ) {
+		log( 'No valid recipients', $email_id, null );
+		return;
+	}
+
+	// Associate attendee to tix_email as a recipient.
+	$result = add_post_meta( $email_id, 'tix_email_recipient_id', $recipient_id );
+
+	if ( ! $result ) {
+		log( 'Failed to add recipients:', $email_id, $recipient_id );
+	} else {
+		log( 'Successfully added recipients:', $email_id, $recipient_id );
+	}
+}
+
+/**
+ * Returns true if an emailed will be sent or is queued.
+ *
+ * @return bool
+ */
+function is_email_already_sent_or_queued( $email_id ) {
+	$email = get_post( $email_id );
+	return 'publish' === $email->post_status || 'pending' === $email->post_status;
+}
+
+/**
+ * Returns true if it is time to send the email.
+ *
+ * @return bool
+ */
+function is_time_to_send_email( $email_id ) {
+	$blog_id       = get_current_blog_id();
+	$wordcamp_post = get_wordcamp_post( $blog_id );
+
+	if ( ! $wordcamp_post ) {
+		log( 'Couldn\'t retrieve wordcamp for blog id:', $email_id, $blog_id );
+		return false;
+	}
+
+	$end_date = $wordcamp_post->meta['End Date (YYYY-mm-dd)'][0];
+
+	if ( empty( $end_date ) ) {
+		// Some sites that are only 1-day events will only have the start date.
+		$end_date = $wordcamp_post->meta['Start Date (YYYY-mm-dd)'][0];
+	}
+
+	$date            = new \DateTime("@$end_date ");
+	$current_date    = new \DateTime();
+	$interval        = $current_date->diff($date);
+	$days_difference = $interval->days;
+
+	return DAYS_AFTER_TO_SEND === $days_difference;
+}
+
+/**
+ * Delete the temporary attendee.
+ */
+function delete_temp_attendee() {
+	$attendee_id = get_wordcamp_attendees_id();
+	wp_delete_post( $attendee_id, true );
+	delete_option( TEMP_ATTENDEE_ID );
+}
+
+/**
+ * Associates recipients to email and changes its status to be picked up
+ * by the camptix email cron job `tix_scheduled_every_ten_minutes`.
+ */
+function queue_organizer_survey() {
+	$email_id = get_email_id();
+
+	if ( empty( $email_id ) ) {
+		return;
+	}
+
+	// check to make sure we didn't already send an email for this wordcamp.
+	if ( is_email_already_sent_or_queued( $email_id ) ) {
+		return;
+	}
+
+	if ( ! is_time_to_send_email( $email_id ) ) {
+		return;
+	}
+
+	add_temp_attendee();
+	associate_attendee_to_email( $email_id );
+
+	$email_status = queue_survey_email( $email_id );
+
+	if ( is_wp_error( $email_status ) ) {
+		log( 'Failed updating email status', $email_id, $email_status );
+	} else {
+		log( 'Email status change to `pending`.', $email_id );
+	}
+
+	// Remove the cron job that queues everything.
+	wp_clear_scheduled_hook( 'wc_organizer_debrief_survey_email' );
```

Comment:
It looks like this will just get added again on the next page load by `Cron\schedule_jobs()`, is that right? if so, is there any advantage to removing it here?